src/cryptoadvance/specter/server_endpoints/wallets.py

```diff
@@ -1000,3 +991,33 @@ def rescan_progress(wallet_alias):
     except SpecterError as se:
         app.logger.error("SpecterError while get wallet rescan_progress: %s" % se)
         return {}
+
+
+@wallets_endpoint.route("/wallet/<wallet_alias>/get_label", methods=["POST"])
+@login_required
+def get_label(wallet_alias):
+    try:
+        wallet = app.specter.wallet_manager.get_by_alias(wallet_alias)
+        address = request.form.get("address", "")
+        label = wallet.getlabel(address)
+        return {
+            "address": address,
+            "label": label,
+        }
+    except SpecterError as se:
+        app.logger.error("SpecterError while get label: %s" % se)
```

Comment:
What about flashing the error to the user? I thought about the SpecterError  as something which is meaningful to the User. We can and imho should raise these kind of errors with that in mind?
Unfortunately I just checked and can't find a single place in the code where this is the case.

Reply:
For the reference: Flashing doesn't make sense for api-calls returning json. So instead of returning empty {}, the approach should be to return a reasonable error,essage and 500 code (or better suited code depending on what you know about the rerror).
Then the client is supposed to deal as best as he can with the situation.